### PR TITLE
Checkin to allow deployments to sonatype's repos

### PR DESCRIPTION
**Pull request by sduskis**
see https://oss.sonatype.org/content/repositories/snapshots/com/google/cloud/bigtable/ for the result
see http://central.sonatype.org/pages/apache-maven.html for more info about the requirements

### CODE REVIEW

File: bigtable-hbase-1.0/pom.xml [hunk 1]
```diff
@@ -66,6 +66,7 @@ limitations under the License.
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
+                <version>2.6</version>
```
Comment (AngusDavis) [inline bigtable-hbase-1.0/pom.xml]: Can this version be specified in the parent pom in pluginManagement?

File: bigtable-hbase/pom.xml [hunk 2]
```diff
@@ -44,28 +44,6 @@ limitations under the License.
         </dependency>
 
         <dependency>
-            <groupId>com.google.api-client</groupId>
```
Comment (AngusDavis) [inline bigtable-hbase/pom.xml]: It's not obvious why this change is here - did closing the snapshot repository with these fail?

File: pom.xml [hunk 3]
```diff
@@ -21,6 +21,21 @@ limitations under the License.
     <artifactId>bigtable-client</artifactId>
     <version>0.1.7-SNAPSHOT</version>
     <packaging>pom</packaging>
+    <name>${project.groupId}:${project.artifactId}</name>
+    <url>https://cloud.google.com/bigtable/</url>
+    <description>
```
Comment (AngusDavis) [inline pom.xml]: This seems like marketing copy instead of "A client to access Google Cloud Bigtable (http://...) from HBase".

It might also make sense to have descriptions in the submodules if the intent is for them to be referenced separately (e.g., the lightweight client in bigtable-grpc).

File: pom.xml [hunk 4]
```diff
@@ -307,6 +356,55 @@ limitations under the License.
...
+                        <execution>
+                          <id>attach-javadocs</id>
+                          <goals>
+                      <goal>jar</goal>
+                        </goals>
+                      </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>org.sonatype.plugins</groupId>
```
Comment (AngusDavis) [inline pom.xml]: This is optional. mvn deploy works fine without it. It's completely up to you, though.

File: pom.xml [hunk 5]
```diff
@@ -307,6 +356,55 @@ limitations under the License.
...
+                    <version>1.6.3</version>
+                    <extensions>true</extensions>
+                    <configuration>
+                        <serverId>ossrh</serverId>
+                        <nexusUrl>https://oss.sonatype.org/</nexusUrl>
+                        <autoReleaseAfterClose>true</autoReleaseAfterClose>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
```
Comment (AngusDavis) [inline pom.xml]: Does this get invoked properly (it's in management and not in a top-level build)? 
Even if it does, would it make sense to add it only in to a release profile so people can mvn install without having GPG setup correctly? An example: https://github.com/GoogleCloudPlatform/bigdata-interop/blob/master/pom.xml#L117

File: bigtable-hbase/pom.xml (hunk 2 above)
Comment (sduskis) [inline bigtable-hbase/pom.xml]: There were 3 duplicated dependencies that mvn deploy complained about.  I had to remove them for mvn deploy to work.

File: pom.xml [hunk 6]
```diff
@@ -276,6 +298,33 @@ limitations under the License.
         <tag>HEAD</tag>
     </scm>
 
+    <developers>
+       <developer>
+         <name>Angus Davis</name>
+         <email>mngusdavis@google.com</email>
+         <organization>Google</organization>
```
Comment (AngusDavis) [inline pom.xml]: s/mngusdavis/angusdavis/ :)

Comment (AngusDavis) [inline pom.xml]: Also - I don't really mind being named, but the other option is a simple:
  <developers>
    <developer>
      <organization>Google Inc.</organization>
      <organizationUrl>http://www.google.com</organizationUrl>
    </developer>
  </developers>

**Comment by AngusDavis** [comment]
If I recall, an issue tracker is also required in the pom.

File: bigtable-hbase/pom.xml (hunk 2 above)
Comment (AngusDavis) [inline bigtable-hbase/pom.xml]: Ah, makes sense. Thanks for the clarification.

File: pom.xml (hunk 5 above)
Comment (sduskis) [inline pom.xml]: A release profile makes sense for both this and sources/javadoc.  I'll make it so.

**Comment by sduskis** [comment]
Updated the top pom.xml to make the javadoc and source jar plugins work as well as the signing.

I still need to update the description.  If that's all there is left, then it would be nice to get this PR merged.

**Comment by sduskis** [comment]
I added more information to the top level description.  I also changed autoReleaseAfterClose to false.  PTAL.

**Comment by AngusDavis** [comment]
Couple of small requests, otherwise LGTM.

**Comment by sduskis** [comment]
Moved the distributionManagement closer to repositories.  Also added basic name/description to all modules.
